### PR TITLE
Limit the size of an index to 1k urls

### PR DESCRIPTION
**Pull request by ahmad-sanad**

### CODE REVIEW

**Comment by academia-feedback** [comment]
__________________________________
Type your response ABOVE THIS LINE to reply

------------------------------------------------------------------------------------------------------------

Re: [academia-edu/simple-sitemap] Limit the size of an index to 1k urls (#4)

Academia.edu | Feb 17, 2017 11:14AM PST

 Thank you for contacting Academia.edu support! This is an auto-reply message indicating we have received your message.  This is an auto-reply message indicating we have received your message. You should receive a response within 48 business hours, though most concerns should be responded to much sooner. 

We also encourage you to review our Help Center at http://support.academia.edu. Many common questions are answered there! Thank you for your patience. 

Thanks in advance for your patience.

------------------------------------------------------------------------------------------------------------
This message was sent to  reply@reply.github.com in reference to Case #216019.
------------------------------------------------------------------------------------------------------------


[[453efe1366d92277407724d8d8d193536302e652-896334439]]

File: lib/simple_sitemap/generators/index.rb [hunk 1]
```diff
@@ -24,21 +25,25 @@ def add_sitemap(name)
...
 
       def write!
-        xml = to_xml
-        index_filename = 'index.xml'
-        if @config.prefix
-          index_filename = "#{@config.prefix}_#{index_filename}"
+        @sitemaps.each_slice(SimpleSitemap::MAX_INDEX_SIZE) do |sitemaps|
+          xml = to_xml(sitemaps)
+          write_file index_filename, xml
+          @index_count += 1
```
Comment (nate00) [inline lib/simple_sitemap/generators/index.rb]: Could you use `with_index` instead of maintaining this index yourself? Otherwise :+1:

**Comment by academia-feedback** [comment]
__________________________________
Type your response ABOVE THIS LINE to reply

------------------------------------------------------------------------------------------------------------

Re: [academia-edu/simple-sitemap] Limit the size of an index to 1k urls (#4)

Academia.edu | Feb 17, 2017 01:41PM PST

 Thank you for contacting Academia.edu support! This is an auto-reply message indicating we have received your message.  This is an auto-reply message indicating we have received your message. You should receive a response within 48 business hours, though most concerns should be responded to much sooner. 

We also encourage you to review our Help Center at http://support.academia.edu. Many common questions are answered there! Thank you for your patience. 

Thanks in advance for your patience.

------------------------------------------------------------------------------------------------------------
This message was sent to  reply@reply.github.com in reference to Case #216019.
------------------------------------------------------------------------------------------------------------


[[453efe1366d92277407724d8d8d193536302e652-896334439]]

File: lib/simple_sitemap/generators/index.rb [hunk 2]
```diff
@@ -25,16 +24,15 @@ def add_sitemap(name)
...
-          @index_count += 1
+          write_file index_filename(index_count), xml
         end
       end
 
-      def index_filename
+      def index_filename(index_count)
         filename = "index.xml"
-        filename = "index_#{@index_count}.xml" if @index_count > 1
+        filename = "index_#{index_count}.xml" if index_count > 1
```
Comment (nate00) [inline lib/simple_sitemap/generators/index.rb]: `index_count > 0`?

Comment (ahmad-sanad) [inline lib/simple_sitemap/generators/index.rb]: Whoops

Comment (adnanakil) [inline lib/simple_sitemap/generators/index.rb]: whoopsy daisy!

**Pull request merged**